### PR TITLE
Add tests focusing specific areas in memory subsystem

### PR DESCRIPTION
**Pull request by harish-24**
Patch adds test which focuses on specific tests in areas on memory subsystem

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Numa:
```
 (1/1) memory/mm_subsystem.py:MmSubsystemTest.test:  FAIL: Test failed, please check above for failures (663.07 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/harish/tests-inject/results/job-2020-08-20T13.06-7ca2e0b/results.html
JOB TIME   : 664.72 s
```
Non-numa:
```
 (1/1) memory/mm_subsystem.py:MmSubsystemTest.test:  FAIL: Test failed, please check above for failures (229.68 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/harish/tests/results/job-2020-08-20T22.31-9f8f2e3/results.html
JOB TIME   : 230.58 s
```

File: memory/mm_subsystem.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            deps.extend(['glibc-static', 'git', 'runc'])
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        git.get_repo("https://gitlab.com/cailca/linux-mm",
+                     destination_dir=self.logdir)
+        os.chdir(self.logdir)
+        if detected_distro.name == "SuSE":
+            suse_patch = self.fetch_asset("1.patch", locations=[
+                "https://patch-diff.githubusercontent.com/raw/harish-24/"
```
Comment (PraveenPenguin) [inline memory/mm_subsystem.py:72]: is this location available as  i am not able to locate via browser and wget also not works 
```
[praveen@MiWiFi-R3L-srv ~]$ wget https://patch-diff.githubusercontent.com/raw/harish-24/linux-mm/pull/.patch
--2020-08-27 13:55:15--  https://patch-diff.githubusercontent.com/raw/harish-24/linux-mm/pull/.patch
Resolving patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)... 13.234.176.102
Connecting to patch-diff.githubusercontent.com (patch-diff.githubusercontent.com)|13.234.176.102|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
Cookie coming from patch-diff.githubusercontent.com attempted to set domain to github.com
2020-08-27 13:55:15 ERROR 404: Not Found.

[praveen@MiWiFi-R3L-srv ~]$ 
```

File: memory/mm_subsystem.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            deps.extend(['libpthread-stubs0-dev', 'git'])
+        elif detected_distro.name == "SuSE":
+            deps.extend(['glibc-devel-static', 'git-core'])
+        else:
+            deps.extend(['glibc-static', 'git', 'runc'])
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        git.get_repo("https://gitlab.com/cailca/linux-mm",
+                     destination_dir=self.logdir)
```
Comment (PraveenPenguin) [inline memory/mm_subsystem.py:68]: is there any specif reason to using logdir ?

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]
@harish-24 thanks for Test, the overall test looks good, just some basic query can you please check those

File: memory/mm_subsystem.py (hunk 1 above)
Comment (harish-24) [reply memory/mm_subsystem.py:72]: Thanks for catching this. It seems to be a typo (missing "1") when editing indentation I guess, fixing it now.

File: memory/mm_subsystem.py (hunk 2 above)
Comment (harish-24) [reply memory/mm_subsystem.py:68]: Would want the testcase to be preserved along with the result as these are lightweight and easily relatable to failures.

File: memory/mm_subsystem.py (hunk 1 above)
Comment (PraveenPenguin) [reply memory/mm_subsystem.py:72]: ah ok sure

File: memory/mm_subsystem.py (hunk 2 above)
Comment (PraveenPenguin) [reply memory/mm_subsystem.py:68]: ok why I said an after run we should have directory clean any I fine with any approach

**Review by PraveenPenguin — APPROVED** [review]

**Comment by harish-24** [comment]
@PraveenPenguin Removed patch applying as the patch is merged upstream

**Pull request merged**